finagle-core/src/main/scala/com/twitter/finagle/tracing/Id.scala

```diff
@@ -49,7 +49,11 @@ object SpanId {
 
   def fromString(spanId: String): Option[SpanId] =
     try {
-      Some(SpanId(new RichU64String(spanId).toU64Long))
+      // Tolerates 128 bit X-B3-TraceId by reading the right-most 16 hex
+      // characters (as opposed to overflowing a U64 and starting a new trace).
+      val length = spanId.length()
+      val lower64Bits = length <= 16 ? spanId : spanId.substring(length - 16)
```

Comment:
in scala, we usually use `if (length <= 16) spanId else spanId.substring(length - 16)` as the ternary operator.